lib/jsonrpc.ml

```diff
@@ -554,27 +554,41 @@ let get name dict =
     raise (Malformed_method_request str)
   | Some v -> v
 
-let call_of_string str =
+let version_and_call_of_string str =
   match of_string str with
   | Dict d ->
-    let name = match get "method" d with String s -> s | _ -> raise (Malformed_method_request str) in
-    let params =
+    let name =
+      match get "method" d with
+      | String s -> s
+      | _ -> raise (Malformed_method_request "method")
```

Comment:
@jonludlam I haven't renamed the function, I only changed its internals and added a new function. If someone already uses the old function and upgrades, their code will still compile, but if they're catching Malformed_method_request and the method suddenly starts throwing malformed_method_error it will be an issue. Or am I missing something?